src/unix/kerberos_gss.cc

```diff
@@ -468,7 +468,7 @@ gss_result authenticate_gss_client_wrap(gss_client_state* state,
         input_token.length = len;
     }
 
-    if (user) {
+    if (user && *user) {
```

Comment:
Should the check in L519 also be adjusted in the same way? The memory management code in this file is honestly a bit hard to follow, but it seems that way to me

Reply:
I'm also having a hard time following.

My suspicion is that, since `user` is never null, that release is never being executed. Or at least has not been since the `ToStringWithNonStringAsEmpty` behavior was introduced.

Then again, I don't understand why we'd only need to release the buffer if there is no user. Maybe the right fix is to remove that condition altogether and just call the release if there is any data in the buffer?

Reply:
> Then again, I don't understand why we'd only need to release the buffer if there is no user

I assume that it corresponds to this condition here, hence my suggestion above – if this branch is taken, then `input_token` contains a stack-allocated reference to `buf`, rather than a dynamically allocated buffer